src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java

```diff
@@ -6,74 +6,123 @@
-                        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().mustBeAPlayer
-                                .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-                        return true;
-                    }
-                    if ((cs.hasPermission(command.permission) || command.permission
-                            .equalsIgnoreCase("") || command.permission
-                            .equalsIgnoreCase("iridiumskyblock.")) && command.enabled) {
-                        command.execute(cs, args);
-                    } else {
-                        // No permission
-                        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().noPermission
-                                .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-                    }
-                    return true;
-                }
-            }
-        } else {
-            if (cs instanceof Player) {
-                Player p = (Player) cs;
+    public boolean onCommand(@NotNull CommandSender commandSender, org.bukkit.command.@NotNull Command cmd, @NotNull String label, String[] args) {
+        if (args.length == 0) {
+            if (commandSender instanceof Player) {
+                Player p = (Player) commandSender;
+                // TODO: Send message
                 return true;
             }
         }
-        cs.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().unknownCommand
+
+        for (Command command : commands) {
+            // We don't want to execute other commands or ones that are disabled
+            if (!(command.aliases.contains(args[0]) && command.enabled)) {
+                continue;
+            }
+
+            if (command.player && !(commandSender instanceof Player)) {
+                // Must be a player
+                commandSender.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().mustBeAPlayer
+                        .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
+                return false;
+            }
+
+            if (!((commandSender.hasPermission(command.permission) || command.permission
+                    .equalsIgnoreCase("") || command.permission
+                    .equalsIgnoreCase("iridiumskyblock.")) && command.enabled)) {
+                // No permissions
+                commandSender.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().noPermission
+                        .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
+                return false;
+            }
+
+            command.execute(commandSender, args);
+            return true;
+        }
+        // Unknown command message
+        commandSender.sendMessage(StringUtils.color(iridiumSkyblock.getMessages().unknownCommand
                 .replace("%prefix%", iridiumSkyblock.getConfiguration().prefix)));
-        return true;
+        return false;
```

Comment:
Why return false?

Reply:
Because you should return false when the command is not valid... It doesn't really matter tho

https://hub.spigotmc.org/javadocs/spigot/org/bukkit/command/CommandExecutor.html#onCommand(org.bukkit.command.CommandSender,org.bukkit.command.Command,java.lang.String,java.lang.String%5B%5D)